task1/Program.cs

```diff
@@ -45,5 +71,792 @@ public static decimal PerformCalculation(int r)
+            {
+                Console.Write("*");
+            }
+            Console.Write("\n");
+        }
+
+        public static void DrawRectangularFrame(int height, int width)
+        {
+            DrawString(width);
+            for (int i = 0; i < height - 2; i++)
+            {
+                for (int k = 0; k < width; k++)
+                {
+                    if (k == 0 || k == width - 1)
+                    {
+                        Console.Write("*");
+                        continue;
+                    }
+                    else
+                    {
+                        Console.Write(" ");
+                    }
+                }
+                Console.Write("\n");
+            }
+            DrawString(width);
+            Console.Write("\n");
+        }
+
+        public static void DrawRectangular(int height, int width)
+        {
+            for (int i = 0; i < height; i++)
+            {
+                for (int k = 0; k < width; k++)
+                {
+                    Console.Write("*");
+                }
+                Console.Write("\n");
+            }
+            Console.Write("\n");
+        }
+
+        public static void CalculatorOfPossibleRouts()
+        {
+            Console.WriteLine("Input number of clients");
+            int numberOfClients = Convert.ToInt32(Console.ReadLine());
+            int i = 1;
+            int result = 1;
+            do
+            {
+                result *= i;
+                i++;
+            } while (i <= numberOfClients);
+            Console.WriteLine($"For {numberOfClients} clients it's possible to create {result} ways of delivery route.");
+        }
+
+        public static void OperationsWithNumbersBtwAB()
+        {
+            Console.WriteLine("Input two numbers, first number < second number");
+            Console.WriteLine("The first number is");
+            int firstNumber = Convert.ToInt32(Console.ReadLine());
+            Console.WriteLine("The second number is");
+            int secondNumber = Convert.ToInt32(Console.ReadLine());
+            int sum = 0;
+            string report = "";
+            for (int i = firstNumber + 1; i < secondNumber; i++)
+            {
+                sum += i;
+                if (i % 2 != 0)
+                {
+                    report += Convert.ToString(i) + ", ";
+                }
+            }
+            Console.WriteLine($"Sum of numbers between {firstNumber} and {secondNumber} is {sum}");
+            Console.WriteLine("Odd numbers are " + report);
+        }
+
+        public static void HowManyEvenCyfresInNumber()
+        {
+            Console.WriteLine("Input number");
+            string inputNumber = Console.ReadLine();
+            int testNumber = Convert.ToInt32(inputNumber);
+            int counter = 0;
+            int temp = testNumber;
+            for (int i = 0; i < inputNumber.Length; i++)
+            {
+                if ((temp - temp / 10 * 10) % 2 == 0)
```

Comment:
I think these calculations are not clear and it would be better if you stored intermediate values in variables with clear names.